### ISSUE TITLE
ArrayPrimitive rule crashes with InvalidFirElementTypeException when analysis has unresolved types — regression from v1

### ISSUE THREAD

**Issue by ryanulep** [detekt/detekt#9362]
## Expected Behavior

detekt should not crash the entire analysis run when a rule encounters a type reference whose type cannot be resolved by FIR.

For the `ArrayPrimitive` rule specifically, unresolved or missing FIR elements should be skipped, similar to how detekt v1 rules using `BindingContext` handled missing types safely:

```kotlin
// v1 — safe when types are missing
val type = bindingContext.getType(typeRef) ?: return false
```

Either:

1. `KaFirTypeProvider.getType()` / the `type` property inside `analyze { }` should return an error type or nullable result instead of throwing when no FIR element exists, consistent with the v1 `BindingContext` behavior; or
2. `ArrayPrimitive.isArrayPrimitive()` should catch `InvalidFirElementTypeException` and return `false`, so the rule skips unresolvable elements rather than crashing analysis.

## Observed Behavior

In detekt v2, `ArrayPrimitive` uses the FIR Analysis API:

```kotlin
private fun KtTypeReference.isArrayPrimitive(): Boolean =
    analyze(this) {
        type.symbol?.classId == StandardClassIds.Array && type.arrayElementType?.isPrimitive == true
    }
```

When a PSI element has no FIR representation, `KaFirTypeProvider.getType()` calls `handleUnexpectedFirElementError` and throws `InvalidFirElementTypeException` instead of returning `null` or an error type.

This causes the entire detekt analysis run to crash when a large module contains unresolvable types, for example from generated code that cannot be placed on the classpath at analysis time.

This appears to be a regression from detekt v1 behavior, where equivalent `BindingContext.getType(...)` calls returned `null` and rules were written to handle that case safely.

Full stack trace:

```text
The original exception message was: unexpected element of type: no element found
Running detekt '2.0.0-alpha.2' on Java '11.0.23+9-LTS' on OS 'Linux'
If the exception message does not help, please feel free to create an issue on our GitHub page.
    at dev.detekt.core.AnalyzerKt.throwIllegalStateException(Analyzer.kt:109)
    at dev.detekt.core.AnalyzerKt.access$throwIllegalStateException(Analyzer.kt:1)
    at dev.detekt.core.Analyzer.runSync(Analyzer.kt:46)
    at dev.detekt.core.Analyzer.run(Analyzer.kt:37)
    at dev.detekt.core.tooling.Lifecycle.analyze$lambda$4(Lifecycle.kt:91)
    at dev.detekt.core.util.PerformanceMonitor.measure(PerformanceMonitor.kt:26)
    at dev.detekt.core.tooling.Lifecycle.measure(Lifecycle.kt:69)
    at dev.detekt.core.tooling.Lifecycle.analyze(Lifecycle.kt:88)
    at dev.detekt.core.tooling.AnalysisFacade.runAnalysis$lambda$0(AnalysisFacade.kt:26)
Caused by: java.lang.AssertionError: unexpected element of type: no element found
    at com.intellij.openapi.diagnostic.DefaultLogger.error(DefaultLogger.java:83)
    at com.intellij.openapi.diagnostic.Logger.error(Logger.java:436)
    at org.jetbrains.kotlin.analysis.api.fir.components.KaFirTypeProvider.handleUnexpectedFirElementError(KaFirTypeProvider.kt:206)
    at org.jetbrains.kotlin.analysis.api.fir.components.KaFirTypeProvider.getType(KaFirTypeProvider.kt:194)
    at org.jetbrains.kotlin.analysis.api.impl.base.KaBaseSession.getType(KaBaseSession.kt)
    at dev.detekt.rules.performance.ArrayPrimitive.isArrayPrimitive(ArrayPrimitive.kt:77)
    at dev.detekt.rules.performance.ArrayPrimitive.reportArrayPrimitives$lambda$0(ArrayPrimitive.kt:71)
    at dev.detekt.rules.performance.ArrayPrimitive$reportArrayPrimitives$$inlined$collectDescendantsOfType$1.invoke(psiUtils.kt:478)
    at dev.detekt.rules.performance.ArrayPrimitive$reportArrayPrimitives$$inlined$collectDescendantsOfType$1.invoke(psiUtils.kt:477)
    at dev.detekt.rules.performance.ArrayPrimitive$reportArrayPrimitives$$inlined$collectDescendantsOfType$2.visitElement(psiUtils.kt:391)
    at dev.detekt.rules.performance.ArrayPrimitive.reportArrayPrimitives(ArrayPrimitive.kt:122)
    at dev.detekt.rules.performance.ArrayPrimitive.visitNamedDeclaration(ArrayPrimitive.kt:65)
Caused by: org.jetbrains.kotlin.analysis.low.level.api.fir.api.InvalidFirElementTypeException: unexpected element of type: no element found
    at org.jetbrains.kotlin.analysis.api.fir.components.KaFirTypeProvider.handleUnexpectedFirElementError(KaFirTypeProvider.kt:205)
```

## Steps to Reproduce

1. Run detekt `2.0.0-alpha.2` in full analysis mode on a Kotlin project/module that contains unresolvable types.
2. Ensure the `performance.ArrayPrimitive` rule is enabled.
3. Include a type reference that FIR cannot resolve or map to a FIR element, such as a type from generated code that is not available on the analysis classpath.
4. Run detekt.

Example scenario:

```kotlin
// GeneratedMissingType is generated in normal builds,
// but is not available on detekt's analysis classpath.

class Example {
    fun foo(value: Array<GeneratedMissingType>) {
        println(value)
    }
}
```

Expected result: detekt skips the unresolved type reference or treats it as non-matching for `ArrayPrimitive`.

Actual result: detekt crashes with `InvalidFirElementTypeException` from `KaFirTypeProvider.getType()`.

## Context

This affects large modules where some generated or otherwise unavailable types cannot be placed on the detekt analysis classpath.

In detekt v1, this did not crash analysis because rules using `BindingContext.getType(...)` received `null` for unresolvable types and could return safely.

In detekt v2, the equivalent FIR Analysis API access throws, and `ArrayPrimitive` is not guarded against that exception. As a result, a single unresolvable type can fail the entire analysis run.

Current workaround:

```yaml
# detekt.yml
performance:
  ArrayPrimitive:
    active: false
```

This avoids the crash but disables the rule entirely.

## Your Environment

* Version of detekt used: `2.0.0-alpha.2`
* Version of Gradle used (if applicable): N/A
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: Linux
* JVM: `11.0.23+9-LTS`
* Kotlin version: `2.2.21`
* Analysis mode: full

**Comment by 3flex** [comment]
Agree that the issue with the rule needs to be fixed, but really the root cause is not having the correct classpath.

You said:

> This affects large modules where some generated or otherwise unavailable types cannot be placed on the detekt analysis classpath.

Why can they not be placed on the analysis classpath? What can we change in DGP to make this work automatically?
